diff --git a/src/x509/x509_minimal_full.c b/src/x509/x509_minimal_full.c
index 2b54426..7cd8a5f 100644
--- a/src/x509/x509_minimal_full.c
+++ b/src/x509/x509_minimal_full.c
@@ -34,7 +34,7 @@ br_x509_minimal_init_full(br_x509_minimal_context *xc,
 	 * Note: the X.509 validation engine will nonetheless refuse to
 	 * validate signatures that use MD5 as hash function.
 	 */
-	static const br_hash_class *hashes[] = {
+	const br_hash_class *hashes[] = {
 		&br_md5_vtable,
 		&br_sha1_vtable,
 		&br_sha224_vtable,
